### DISCUSSION TITLE
switchbot, http-shortcuts and macrodroid

### DISCUSSION THREAD

**Discussion by walty68** [Waboodoo/HTTP-Shortcuts#417] [Q&A]
i try to use api of switchbot for learning on watch the temperature of a sensor of switchbot, but i cant do that because i am really new of do it, anyone can help me?

**Comment by Waboodoo** [comment]
Hi,
I've never worked with any switchbot devices, but I found this API documentation: https://github.com/OpenWonderLabs/SwitchBotAPI?tab=readme-ov-file#introduction
You would need to create an API token as described in the documentation above, and then you can construct a shortcut which calls the API at `https://api.switch-bot.com`.
Maybe this will help you to get started.

The trickiest bit might be the authentication. You'll need to compute some values, for which you need the [Scripting feature](https://http-shortcuts.rmy.ch/scripting) to write some JavaScript code and store the computed values into [variables](https://http-shortcuts.rmy.ch/variables) so that you can use them in custom headers.
